tslearn/preprocessing.py

```diff
@@ -146,11 +147,10 @@ def fit_transform(self, X, **kwargs):
             Rescaled time series dataset
         """
         X_ = to_time_series_dataset(X)
-        for i in range(X_.shape[0]):
-            for d in range(X_.shape[2]):
-                cur_mean = X_[i, :, d].mean()
-                cur_std = X_[i, :, d].std()
-                if cur_std == 0.:
-                    cur_std = 1.
-                X_[i, :, d] = (X_[i, :, d] - cur_mean) * self.std_ / cur_std + self.mu_
+        mean_t = numpy.mean(X_, axis=1)[:, numpy.newaxis, :]
```

Comment:
I tried converting the code to numpy in the past and discovered that it actually ran slower than a Pure-Python loop. Maybe this should be tested more extensively, and if it indeed runs slower, convert back to the old code

Reply:
@GillesVandewiele : is it still slower for datasets with many time series? Because it definitely looks like a better solution than the original one in theory :)

Reply:
I mentioned it in Issue #52 right before getting closed, so maybe you looked over it. My code was not exactly the same as this:

```
X_ = to_time_series_dataset(X)
means, stds = np.mean(X_, axis=1), np.std(X_, axis=1)
stds[stds == 0] = 1
return (X_ - means) * self.std_ / stds + self.mu_
```

This seemed to run slower when I did some simple tests, but maybe should be double checked. It is definitely cleaner code!

Reply:
This conversion to numpy can btw also easily be done for the MinMaxScaler!

Reply:
Yep, but in Issue #52, you tested on a single long time series, which did not benefited from removing the loop. So hopefully with many time series that would be better!

Test:
```
time_series = numpy.random.randn(100000, 100, 10)
t0 = time.time()
TimeSeriesScalerMinMax().fit_transform(time_series)
t1 = time.time()
TimeSeriesScalerMinMaxTuned().fit_transform(time_series)
t2 = time.time()
```

With `TimeSeriesScalerMinMaxTuned` being the proposed implementation.

Gives:
```
12.474971771240234 5.250811815261841
```

So, good idea :)

Reply:
You are right! Mistake from my side! I benched it myself now as well :)  

![numpy_times](https://user-images.githubusercontent.com/5750603/44269192-1a6ab500-a234-11e8-876e-dd59823fbf00.png)

![python_times](https://user-images.githubusercontent.com/5750603/44269199-1fc7ff80-a234-11e8-8be1-92f82fbb8d7c.png)